jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -495,46 +508,48 @@ public String toString()
 
     private class DumpKeys extends Invocable.NonBlocking
     {
-        private final CountDownLatch latch = new CountDownLatch(1);
-        private final List<Object> _dumps;
-
-        private DumpKeys(List<Object> dumps)
-        {
-            this._dumps = dumps;
-        }
-
+        private final Exchanger<List<String>> _dump = new Exchanger<>();
```

Comment:
I don't like using an `Exchanger` instead of a `CountDownLatch`.
The reason is that the exchanger may block the thread that runs the `DumpKeys` action (see below, 500 ms hardcoded), while `CountDownLatch` never blocks it.

Reply:
So may grabbing the lock above and we do that.

Reply:
So may grabbing the lock above and we do that.

Reply:
What lock ? The previous implementation was just calling `CountDownLatch.countDown()`, no locks, no wait, nothing.
The JMX thread was waiting, not the selector thread. Now we have the selector thread possibly waiting 500 ms if the JMX thread is for some reason late to arrive at the exchanger (GC, scheduling, etc.).

Reply:
Submitting the dumpkeys class grabs the lock